editor.planx.uk/src/api/upload.ts

```diff
@@ -14,21 +12,16 @@ async function uploadPublicFile(
 ) {
   const { data } = await handleUpload(file, { onProgress, path: "public" });
 
-  return `${process.env.REACT_APP_API_URL}/file/public/${encodeS3URI(
-    data.key
-  )}`;
+  return data.fileUrl;
 }
 
 async function uploadPrivateFile(
   file: any,
   { onProgress }: { onProgress?: (p: any) => void } = {}
 ) {
   const { data } = await handleUpload(file, { onProgress, path: "private" });
-  const passport = useStore.getState().computePassport();
 
-  return useStagingUrlIfTestApplication(passport)(
-    `${process.env.REACT_APP_API_URL}/file/private/${encodeS3URI(data.key)}`
-  );
+  return data.fileUrl;
```

Comment:
I'm not sure I understand why this isn't now ```return useStagingUrlIfTestApplication(passport)(data.fileUrl)``` ?

Reply:
This is a good question, and important to be clear on 👍 

`handleUpload()` still conditionally routes private uploads to the staging API if the user is "test test".

So the return value here will be coming from the staging API if the user is "test test" and will already be constructed correctly.

Reply:
that makes sense, thank you!!